### ISSUE TITLE
Cannot install with PYTHONOPTIMIZE=2

### ISSUE THREAD

**Issue by Yajo** [erocarrera/pydot#145]
Steps to reproduce:

```
$ docker run -e PYTHONOPTIMIZE=2 -it --rm python:2-alpine pip install pydot
Collecting pydot
  Downloading pydot-1.2.3.tar.gz
    Complete output from command python setup.py egg_info:
    pydot.py:33: UserWarning: Couldn't import dot_parser, loading of dot files will not be possible.
      "Couldn't import dot_parser, "
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-eqA_Ct/pydot/setup.py", line 32, in <module>
        long_description="\n".join(pydot.__doc__.split('\n')),
    AttributeError: 'NoneType' object has no attribute 'split'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-eqA_Ct/pydot/
```

When using [PYTHONOPTIMIZE=2](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONOPTIMIZE), [`__doc__` are removed from the generated `*.pyo` files](https://docs.python.org/3/using/cmdline.html#cmdoption-OO), assuming that in a production deployment you never need the docstrings.

Could you please use an alternative method to get that docstring and make it work with maximum optimization?

Thanks!

**Comment by johnyf** [comment]
Thank you for noting this behavior. The plan is to load the long description from the README file, so it should address this problem by not accessing a module docstring. However, there are situations (not necessarily within `pydot`) where docstrings are necessary, for example for parser generators that use `ply`.

**Comment by Yajo** [comment]
But do you mean in a production environment?

**Comment by johnyf** [comment]
To be more accurate, [`ply`](https://github.com/dabeaz/ply) reads production rules defined within docstrings and produces a parser. It can save parser tables for later use, as an optimization. I am not sure whether `ply` can be configured to use those tables directly, without the code that defines the production rules.

In any case, `ply` is not used here (it could be in the future, if `pyparsing` was replaced, see https://github.com/erocarrera/pydot/issues/132#issuecomment-250311252). I mentioned this case to suggest that there are use cases where docstrings would have a function within a production environment.

**Comment by Yajo** [comment]
Oh of course they have. That's why we have `PYTHONOPTIMIZE=1` too 😉.

It's just that, if is it possible to optimize a little bit more, then it's indeed desirable. And otherwise, it would be good to document that.

**Comment by johnyf** [comment]
Addressed by 526f1386eeb253c8d581a2ddb4c1f3602cd50d30, f374f66b93ec7e6bb7a6812abdb2d8a62db879f8.
